examples/Acoustics/compute_spectrum_production.py

```diff
@@ -0,0 +1,280 @@
+"""
+
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+# Input corresponding environmental conditions
+GRID_UNIT = 1  # m
+SPEED_OF_SOUND = 340  # m/s, freestream
+RHO = 1.225  # kg/m^3, freestream
+REFERENCE_SOUND_PRESSURE = 20e-6  # Pa
+
+# Constants used for A-weighting
+K1 = 2.243e16
+K3 = 1.562
+F1 = 20.599
+F2 = 107.653
+F3 = 737.862
+F4 = 12194.220
+
+
+def read_csv_file(file_path):
+    """This function reads in the input file"""
+
+    df = pd.read_csv(file_path, skipinitialspace=True)
+    df = df.iloc[:, :-1]
+    data = df.to_dict(orient="list")
+    return data
+
+
+def apply_windowing_function(acoustics_data, i_observer, window):
+    """This function reads in the CSV input file and returns dict"""
+
+    pressure = np.asarray(acoustics_data[f"observer_{i_observer}_pressure"])
+    indices = np.nonzero(pressure)
+    non_zero_pressure = pressure[indices]
+    n = len(non_zero_pressure)  # number of time points in the time history
+
+    # window correction."None"= No window, "Hann"= Hann Window
+    if window == "Hann":
+        window_function = np.hanning(n)
+        fc = np.sqrt(n / sum(window_function**2))
+    else:
+        window_function = 1.0
+        fc = 1.0
+    # The windows that are actually applied to the time history are
+    # the window functions multiplied by their respective scaling factor
+    windowed_pressure = window_function * fc * non_zero_pressure
+    return windowed_pressure
+
+
+def compute_a_weight(f_m):
+    """This function computes the a_weight"""
+
+    w_cm = K1 * f_m**4 / ((f_m**2 + F1**2) ** 2 * (f_m**2 + F4**2) ** 2)
+    w_am = w_cm * K3 * f_m**4 / ((f_m**2 + F2**2) * (f_m**2 + F3**2))
+    return w_am
+
+
+def compute_spectrum(windowed_pressure, time_step_size, weighting):
+    """This function performs the FFT on the pressure signal and computes the OASPL and SPL spectrum"""
+
+    n = len(windowed_pressure)
+    physical_pressure = windowed_pressure * RHO * SPEED_OF_SOUND**2  # Pa
+    physical_time_step_size = time_step_size * GRID_UNIT / SPEED_OF_SOUND  # s
+    frequency_step_size = 1.0 / (n * physical_time_step_size)
+    m = int(np.floor(n / 2) + 1)
+    nyquist_frequency = np.floor(n / 2) * frequency_step_size
+    frequency = np.linspace(0, nyquist_frequency, m)
+    spectrum = np.fft.rfft(physical_pressure)
+    p_c = np.zeros(m, dtype=complex)
+    p_c[0] = spectrum[0] * 1 / n
+    p_c[1:m] = spectrum[1:m] * 2 / n
+
+    if weighting == "A":
+        weighted_mean_square_pressure = np.zeros(m)
+        for k in range(0, m):
+            f_m = frequency[k]
+            w_am = compute_a_weight(f_m)
+            weighted_mean_square_pressure[k] = 0.5 * w_am * np.abs(p_c[k]) ** 2
+        spl = 10 * np.log10(
+            weighted_mean_square_pressure / REFERENCE_SOUND_PRESSURE**2
+        )
+        oaspl = 10 * np.log10(
+            sum(weighted_mean_square_pressure) / REFERENCE_SOUND_PRESSURE**2
+        )
+        if np.isnan(oaspl) == 0:
```

Comment:
```suggestion
        if np.isnan(oaspl) is False:
```

Reply:
there are a number of these similar statements throughout that can be updated

Reply:
This doesn't work. isnan return 0, 1

Reply:
hmm, I receive a boolean True/False. Also here in numpy docs, https://numpy.org/doc/stable/reference/generated/numpy.isnan.html. Maybe a numpy version difference?

Reply:
== False works. 
is False and == False do not do the same thing according to this thread:
https://stackoverflow.com/questions/4591125/is-it-safe-to-replace-with-is-to-compare-boolean-values

Reply:
I see that I confused this point also... From your link above, the appropriate syntax should be:

`if not np.isnan(oaspl):`

Our use of either `== 0` or `is False` are not ideal.

Reply:
Agreed! Sounds good!